danmaku/ui/commonMain/DanmakuFilterConfig.kt

```diff
@@ -0,0 +1,44 @@
+package me.him188.ani.danmaku.ui
+
+import androidx.compose.runtime.Immutable
+import androidx.compose.runtime.Stable
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.Transient
+import me.him188.ani.danmaku.api.Danmaku
+import java.util.UUID
+
+/**4
+ * Defines which [Danmaku] can be displayed.
+ */
+@Immutable
+@Serializable
+data class DanmakuFilterConfig(
+
+    /**
+     * Defines which [Danmaku] can be displayed according to a list of regular expression.
+     */
+    
+    val danmakuRegexFilterList: List<DanmakuRegexFilter> = mutableListOf(),
+    @Suppress("PropertyName") @Transient val _placeholder: Int = 0,
+
+
+    ) {
+    companion object {
+        @Stable
+        val Default = DanmakuFilterConfig()
+    }
+}
+
+@Serializable
+data class DanmakuRegexFilter (
+    val instanceID: String,
+    val name: String = "",
+    val re: String = "",
```

Comment:
`re` is not intuitive enough to me

```suggestion
    val regex: String = "",
```
or `expr`, `expression`.